home/private_dot_local/lib/agent-equipment/agent_equipment/source_resolution.py

```
@@ -0,0 +1,1019 @@
+def _admit_equipment_list(value: object, field: str) -> tuple[str, ...]:
+    if type(value) is not tuple or not value or len(value) > MAX_AVAILABLE_EQUIPMENT:
+        raise ValueError(f"source manifest {field} must be a bounded nonempty list")
+    admitted = tuple(
+        _require_equipment_identity(item, f"{field} identity") for item in value
+    )
+    if admitted != tuple(sorted(set(admitted))):
+        raise ValueError(f"source manifest {field} must be sorted and unique")
+    return admitted
+
+
+def _admit_native_version(
```

Ruff (0.16.1)

[warning] Too many branches (15 > 12)

(PLR0912)